subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/release/tasks/UploadGists.java

```diff
@@ -0,0 +1,64 @@
+package org.shipkit.internal.gradle.release.tasks;
+
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.json.simple.JsonObject;
+import org.json.simple.Jsoner;
+import org.shipkit.internal.gradle.util.FileUtil;
+import org.shipkit.internal.notes.util.IOUtil;
+import org.shipkit.internal.util.GitHubApi;
+
+import java.io.File;
+
+public class UploadGists {
+
+    private static final Logger LOG = Logging.getLogger(UploadGistsTask.class);
+
+    public void uploadGists(UploadGistsTask uploadGistsTask) {
+        uploadGists(uploadGistsTask, new GitHubApi(uploadGistsTask.getGitHubApiUrl(), uploadGistsTask.getGitHubWriteToken()));
+    }
+
+    public void uploadGists(UploadGistsTask uploadGistsTask, GitHubApi gitHubApi) {
+        for (String logFilePattern : uploadGistsTask.getFilesPatterns()) {
+            uploadForPattern(logFilePattern, gitHubApi, uploadGistsTask.getRootDir());
+        }
+    }
+
+    private void uploadForPattern(final String logFilePattern, GitHubApi gitHubApi, String rootDir) {
+        for (String logFile : FileUtil.findFilesByPattern(rootDir, logFilePattern)) {
+            uploadSingleFile(logFile, gitHubApi);
+        }
+    }
+
+    private void uploadSingleFile(String filePath, GitHubApi gitHubApi) {
```

Comment:
For clarity and easy unit testing, I suggest to push all GitHub JSON + posting logic to a separate class. It could expose a single method that takes Strings as parameters (url, token, gist content).